src/main/java/org/jabref/gui/util/comparator/NumericFieldComparator.java

```diff
@@ -10,51 +12,54 @@
 
     @Override
     public int compare(String val1, String val2) {
-        // We start by implementing the comparison in the edge cases (if one of the values is null).
-        if (val1 == null && val2 == null) {
-            return 0;
-        }
+        Integer valInt1 = parseInt(val1);
+        Integer valInt2 = parseInt(val2);
 
-        if (val1 == null) {
+        if (valInt1 == null && valInt2 == null) {
+            if (val1 != null && val2 != null) {
+                return val1.compareTo(val2);
+            } else {
+                return 0;
+            }
+        } else if (valInt1 == null) {
             // We assume that "null" is "less than" any other value.
             return -1;
-        }
-
-        if (val2 == null) {
+        } else if (valInt2 == null) {
             return 1;
         }
 
-        // Now we start the conversion to integers.
-        Integer valInt1 = null;
-        Integer valInt2 = null;
-        try {
-            // Trim in case the user added an unnecessary white space (e.g. 1 1 instead of 11).
-            valInt1 = Integer.parseInt(val1.trim());
-        } catch (NumberFormatException ignore) {
-            // do nothing
+        // If we arrive at this stage then both values are actually numeric !
+        return valInt1 - valInt2;
+    }
+
+    private static Integer parseInt(String number) {
+        if (!isNumber(number)) {
+            return null;
         }
+
         try {
-            valInt2 = Integer.parseInt(val2.trim());
+            return Integer.valueOf(number.trim());
         } catch (NumberFormatException ignore) {
-            // do nothing
+            return null;
         }
+    }
 
-        if (valInt1 == null && valInt2 == null) {
-            // None of the values were parsed (i.e both are not numeric)
-            // so we will use the normal string comparison.
-            return val1.compareTo(val2);
+    private static boolean isNumber(String number) {
+        if (StringUtil.isNullOrEmpty(number)) {
+            return false;
         }
-
-        if (valInt1 == null) {
-            // We assume that strings "are less" than integers
-            return -1;
+        if (number.length() == 1 && number.charAt(0) == '-' && number.charAt(0) == '+') {
```

Comment:
If I haven't misunderstood the purpose of this line it should check the same condition as line 56 (`(c == '-' || c == '+')`)

Reply:
It only filters strings like that '-', '+'. Line 56 allow the first character to be '-' or '+'